### PR TITLE
feature(diffing): foreign key diffing

### PR DESCRIPTION
**Pull request by Scrunshes**
Foreign key diffing was missing so here it is.

### CODE REVIEW

File: src/SnapshotManager.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export class SnapshotManager {
   /**
    * @type {{}}
    */
-  private config: {snapshotDirectory: string, devSnapshotDirectory: string};
+  private config: { snapshotDirectory: string, devSnapshotDirectory: string };
```
Comment (RWOverdijk) [inline src/SnapshotManager.ts]: Could you remove the spaces again?

File: src/SnapshotManager.ts [hunk 2]
```diff
@@ -218,6 +218,16 @@ export class SnapshotManager {
         let field         = newMapping[entity].fields[property];
 
         if (field.relationship) {
+          let previousJoinColumn = previousField.joinColumn;
+          let joinColumn         = field.joinColumn;
+
+          if (fieldChanged(previousJoinColumn, joinColumn)) {
+            let relation = newMapping[entity].relations[field.name];
+            let alter    = getAlterInstructions(instructions.alter, tableName);
+            alter.dropForeign.push(previousJoinColumn.name);
```
Comment (RWOverdijk) [inline src/SnapshotManager.ts]: Newline after declaration group

File: src/SnapshotManager.ts [hunk 3]
```diff
@@ -471,6 +481,17 @@ export class SnapshotManager {
       instructions.drop.push(mapping.fields[property].joinTable.name);
     }
 
+    function createForeign(mapping, property, targetMapping) {
+      let joinColumn = mapping.fields[property].joinColumn;
+      return {
```
Comment (RWOverdijk) [inline src/SnapshotManager.ts]: Newline before return

File: src/SnapshotManager.ts [hunk 4]
```diff
@@ -471,6 +481,17 @@ export class SnapshotManager {
       instructions.drop.push(mapping.fields[property].joinTable.name);
     }
 
+    function createForeign(mapping, property, targetMapping) {
```
Comment (RWOverdijk) [inline src/SnapshotManager.ts]: Can't we reuse an existing method somewhere? It looks like something else in the file.

**Comment by RWOverdijk** [comment]
@Scrunshes In general, good PR! Instead of skipping relationships, you diff them and if needed drop / recreate them. This seems pretty safe to me, well done.

I'm thinking of giving some other parts of the orm a tiny makeover, so I think we'd best use this in a prerelease so we can test it better.

File: src/SnapshotManager.ts (hunk 4 above)
Comment (Scrunshes) [inline src/SnapshotManager.ts]: We can use that function in an other function which does more than creating foreigns, but I figured I would change the least things possibles not to break anything for the moment.

**Comment by RWOverdijk** [comment]
I didn't forget about this, it'll be a release soon, I promise. :)

**Pull request merged**